### PR TITLE
Implement authorisation based on Google Group membership

### PR DESCRIPTION
**Pull request by cb372**
Currently disabled in all environments while we sort out the groups we actually want to use for authorisation

### CODE REVIEW

File: app/AppComponents.scala [hunk 1]
```diff
@@ -1,34 +1,86 @@
...
+import router.Routes
+
+trait ConfigUtil {
+  def configuration: Configuration
+
+  def mandatoryConfig(key: String): String = configuration.getString(key).getOrElse(sys.error(s"Missing config key: $key"))
+
+}
+
+trait GoogleAuthComponents extends ConfigUtil {
```
Comment (fleetwoods) [inline app/AppComponents.scala]: Abstract out?

File: app/AppComponents.scala [hunk 2]
```diff
@@ -1,34 +1,86 @@
...
+        GoogleServiceAccount(
+          email = credentials.getServiceAccountId,
+          privateKey = credentials.getServiceAccountPrivateKey,
+          impersonatedUser = mandatoryConfig("google.groupsAuthorisation.impersonatedUser")
+        )
+      }
+
+      val groupChecker = new GoogleGroupChecker(serviceAccount)
+
+      Logger.info("Google Groups authorisation is ENABLED")
```
Comment (fleetwoods) [inline app/AppComponents.scala]: Output at startup rather than every auth?

File: app/AppComponents.scala (hunk 1 above)
Comment (cb372) [inline app/AppComponents.scala]: Sorry, not quite sure what you mean

File: app/AppComponents.scala (hunk 2 above)
Comment (cb372) [inline app/AppComponents.scala]: Yes, that's how this works. The `groupsAuthConfig` method is only called once at startup, not on every request.

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 71](https://circleci.com/gh/ovotech/comms-audit-log/71) at 2017-02-03 14:53 GMT.

**Pull request merged**